generators/client-2/templates/src/main/webapp/app/admin/user-management/_user-management.state.ts

```diff
@@ -1,152 +1,154 @@
-                    }
-                }
-            }).result.then(function() {
-                $state.go('user-management', null, { reload: true });
-            }, function() {
-                $state.go('user-management');
-            });
-        }]
-    })
-    .state('user-management.edit', {
-        url: '/{login}/edit',
-        data: {
-            authorities: ['ROLE_ADMIN']
-        },
-        onEnter: ['$stateParams', '$state', '$uibModal', function($stateParams, $state, $uibModal) {
-            $uibModal.open({
-                templateUrl: 'app/admin/user-management/user-management-dialog.html',
-                controller: 'UserManagementDialogController',
-                controllerAs: 'vm',
-                backdrop: 'static',
-                size: 'lg',
-                resolve: {
-                    entity: ['User', function(User) {
-                        return User.get({login : $stateParams.login});
-                    }]
-                }
-            }).result.then(function() {
-                $state.go('user-management', null, { reload: true });
-            }, function() {
-                $state.go('^');
-            });
-        }]
-    })
-    .state('user-management.delete', {
-        url: '/{login}/delete',
-        data: {
-            authorities: ['ROLE_ADMIN']
-        },
-        onEnter: ['$stateParams', '$state', '$uibModal', function($stateParams, $state, $uibModal) {
-            $uibModal.open({
-                templateUrl: 'app/admin/user-management/user-management-delete-dialog.html',
-                controller: 'UserManagementDeleteController',
-                controllerAs: 'vm',
-                size: 'md',
-                resolve: {
-                    entity: ['User', function(User) {
-                        return User.get({login : $stateParams.login});
-                    }]
-                }
-            }).result.then(function() {
-                $state.go('user-management', null, { reload: true });
-            }, function() {
-                $state.go('^');
-            });
+    <%_ } _%>
+
+};
+export const userMgmtDetailState = {
+    name: 'user-management-detail',
+    parent: 'admin',
+    url: '/user/:login',
+    data: {
+        authorities: ['ROLE_ADMIN'],
+        pageTitle: 'user-management.detail.title'
+    },
+    views: {
+        'content@': { component: UserMgmtDetailComponent }
+    },
+    resolve: {
+        translatePartialLoader: ['$translate', '$translatePartialLoader', function ($translate, $translatePartialLoader) {
+            $translatePartialLoader.addPart('user-management');
+            return $translate.refresh();
         }]
-    });
-}
+    }
+};
+export const userMgmtNewState = {
+    name: 'user-management.new',
+    url: '/new',
+    data: {
+        authorities: ['ROLE_ADMIN']
+    },
+    // onEnter: ['$stateParams', '$state', '$uibModal', function($stateParams, $state, $uibModal) {
+    //     $uibModal.open({
```

Comment:
@ruddell there has been a release of ng-bootstrap which adds thecomponenet support. so we can workaround by passing components and moving the data fetching logic to onInit of the componenets
I'll upgrade ng-bootstrap now